optimum/intel/openvino/modeling_decoder.py

```diff
@@ -249,6 +251,67 @@ class OVModelForCausalLM(OVBaseDecoderModel, GenerationMixin):
     export_feature = "text-generation"
     auto_model_class = AutoModelForCausalLM
 
+    def __init__(
+        self,
+        model: openvino.runtime.Model,
+        config: PretrainedConfig = None,
+        device: str = "CPU",
+        dynamic_shapes: bool = True,
+        ov_config: Optional[Dict[str, str]] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        **kwargs,
+    ):
+        model = self._try_modify_pastkv_to_lowprecision(model, ov_config, device, **kwargs)
+        super().__init__(model, config, device, dynamic_shapes, ov_config, model_save_dir, **kwargs)
+
+    def _try_modify_pastkv_to_lowprecision(
+        self, model: openvino.runtime.Model, ov_config: Optional[Dict[str, str]] = None, device: str = "CPU", **kwargs
+    ):
+        device = device.upper()
+        pastkv_will_use = core.get_property(device, "INFERENCE_PRECISION_HINT")
+        # ov_config["INFERENCE_PRECISION_HINT"] may override the prefer precision
+        if ov_config:
+            user_precision_hint = ov_config.get("INFERENCE_PRECISION_HINT", "")
+            user_mode_hint = ov_config.get("EXECUTION_MODE_HINT", "")
+            if user_precision_hint in STR_TO_OV_TYPE:
+                pastkv_will_use = STR_TO_OV_TYPE[user_precision_hint]
+            if user_mode_hint.upper() == "ACCURACY":
+                pastkv_will_use = Type.f32
+
+        use_cache = kwargs.get("use_cache", True)
+        has_pastkv = any("past_key_values" in key.get_any_name() for key in model.inputs)
```

Comment:
here you only need `has_pastkv`  (that can be renamed to `use_cache`) as both will always have the same value (see https://github.com/huggingface/optimum-intel/blob/v1.9.3/optimum/intel/openvino/modeling_decoder.py#L130)

Reply:
Fixed.